### PR TITLE
Upgrade to Vert.x 4.4.1, Java 11, Gradle 7.2

### PR DESCRIPTION
**Pull request by julianladisch**
Futurize code.
Add error handling.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -17,6 +17,3 @@ application {
   mainClassName = "io.vertx.howtos.httpclient.JokeVerticle"
 }
 
-tasks.wrapper {
-  gradleVersion = "5.6.3"
-}
```
Comment (tsegismont) [inline build.gradle.kts]: How about changing this to `gradleVersion = "7.2"` instead of removing?

File: pom.xml [hunk 2]
```diff
@@ -7,12 +7,13 @@
 
   <groupId>io.vertx.howtos</groupId>
   <artifactId>http-client-howto</artifactId>
-  <version>1.0-SNAPSHOT</version>
+  <version>1.1.0-SNAPSHOT</version>
```
Comment (tsegismont) [inline pom.xml]: This is not used in any way, you can stick to the default version number

**Review by tsegismont — COMMENTED** [review]
Thank you @julianladisch

File: build.gradle.kts (hunk 1 above)
Comment (julianladisch) [reply build.gradle.kts]: Agreed.

File: pom.xml (hunk 2 above)
Comment (julianladisch) [reply pom.xml]: Reverted.

**Review by tsegismont — APPROVED** [review]
LGTM

**Pull request merged**